test/unit/connection/result/result_test_timestamp.js

```diff
@@ -140,5 +141,238 @@ describe('Result: test timestamp', function () {
           done();
         }
       );
-    });
+    }
+  );
+
+  it('select dateadd(ms,999, to_timestamp_ntz(\'2024-04-16T14:57:58\', \'YYYY-MM-DD"T"HH24:MI:SS\')) AS C1;',
```

Comment:
Can you also add a test with millsecond precision? Ie. `2024-04-16T14:57:58.999`?

Reply:
And check as well higher precision of nanos

Reply:
I have added the requested tests, and cleaned the code somewhat.

I also removed a couple tests that were using `DATEADD` and which were redundant. For each test that used `DATEADD`, now there is a test that parses the corresponding timestamp instead (so that the tests are more readable).

For example, instead of `select dateadd(ms,999, to_timestamp_ntz(\'2024-04-16T14:57:58\', \'YYYY-MM-DD"T"HH24:MI:SS\')) AS C1;` the following expression is used `select  to_timestamp_ntz(\'2024-04-16T14:57:58:999\', \'YYYY-MM-DD"T"HH24:MI:SS:FF3\') AS C1;`